src/context/auth.js

```diff
@@ -13,17 +13,26 @@ const AuthContext = createContext();
 
 const AuthProvider = ({ children }) => {
   const navigate = useNavigate();
-  const location = useLocation();
   const [token, setToken] = useState(null);
 
   useEffect(() => {
     const storedToken = localStorage.getItem('token');
-
     if (storedToken) {
       setToken(storedToken);
-      navigate(location.state?.from?.pathname || '/');
```

Comment:
This line is the BUG with navigation. If we have a stored token AND no redirect path, then we should do nothing. If we do have a redirect path, we might want to redirect.